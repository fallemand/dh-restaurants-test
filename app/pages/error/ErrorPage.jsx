import React from 'react';
import PropTypes from 'prop-types';
import './_error.scss';

const ErrorPage = ({ title, message }) => (
  <div className="error">
    <h1 className="error__title">{title}</h1>
    <h2 className="error__subtitle">{message}</h2>
    <button
      className="error__button"
      type="button"
      onClick={() => window.location.reload()}
    >
      Retry!
    </button>
  </div>
);

ErrorPage.propTypes = {
  title: PropTypes.string,
  message: PropTypes.string,
};

ErrorPage.defaultProps = {
  title: 'Error :(',
  message: 'An error ocurred! Please try again later.',
};

export default ErrorPage;
