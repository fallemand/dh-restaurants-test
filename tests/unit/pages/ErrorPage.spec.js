import React from 'react';
import ErrorPage from '../../../app/pages/error/ErrorPage';

describe('ErrorPage', () => {
  const props = {
    history: { push: jest.fn() },
    match: { params: { id: 1 } },
  };

  global.window = { location: { reload: jest.fn() } };

  it('render component', () => {
    const component = shallow(<ErrorPage />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('Button should redirect to Restaurants List', () => {
    const component = shallow(<NotFoundPage {...props} />);
    component.find('.not-found__button').simulate('click');
    expect(global.window.location.reload).toHaveBeenCalled();
  });
});
