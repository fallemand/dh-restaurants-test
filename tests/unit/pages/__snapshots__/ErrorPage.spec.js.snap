// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage render component 1`] = `
"<div className='error'>
  <h1 className='error__title'>
    Error :(
  </h1>
  <h2 className='error__subtitle'>
    An error ocurred! Please try again later.
  </h2>
  <button className='error__button' type='button' onClick={[Function: onClick]}>
    Retry!
  </button>
</div>"
`;
